apps/emqx_retainer/src/emqx_retainer_mnesia.erl

```diff
@@ -506,10 +499,15 @@ batch_read_number() ->
 reindex(NewIndices, Force, StatusFun) when
     is_boolean(Force) andalso is_function(StatusFun, 1)
 ->
+    %% Do not run on replicants
+    core = mria_rlog:role(),
```

Comment:
`mria:transaction` already routes operations to cores, so this shouldn't be needed?

Reply:
[See the related comment](https://github.com/emqx/emqx/pull/9372#discussion_r1040838267)